wix/Build.OpenJDK_generic.cmd

```diff
@@ -124,26 +126,33 @@ FOR %%A IN (%ARCH%) DO (
 				GOTO FAILED
 			)
 		) ELSE (
-			SET REPRO_DIR=.\SourceDir\!PRODUCT_SKU!!PRODUCT_MAJOR_VERSION!\!PACKAGE_TYPE!\!FOLDER_PLATFORM!\jdk-%PRODUCT_MAJOR_VERSION%+%PRODUCT_PATCH_VERSION%
+			SET REPRO_DIR=.\SourceDir\!PRODUCT_SKU!!PRODUCT_MAJOR_VERSION!\!PACKAGE_TYPE!\!FOLDER_PLATFORM!\jdk-%PRODUCT_MAJOR_VERSION%+%PRODUCT_BUILD_NUMBER%
 			IF !PRODUCT_CATEGORY! == jre (
 				SET REPRO_DIR=!REPRO_DIR!-!PRODUCT_CATEGORY!
 			)
 			IF NOT EXIST "!REPRO_DIR!" (
 				ECHO Second !REPRO_DIR! not exists
-				SET REPRO_DIR=.\SourceDir\!PRODUCT_SKU!!PRODUCT_MAJOR_VERSION!\!PACKAGE_TYPE!\!FOLDER_PLATFORM!\jdk-%PRODUCT_MAJOR_VERSION%.%PRODUCT_MINOR_VERSION%.%PRODUCT_MAINTENANCE_VERSION%+%PRODUCT_PATCH_VERSION%
+				SET REPRO_DIR=.\SourceDir\!PRODUCT_SKU!!PRODUCT_MAJOR_VERSION!\!PACKAGE_TYPE!\!FOLDER_PLATFORM!\jdk-%PRODUCT_MAJOR_VERSION%.%PRODUCT_MINOR_VERSION%.%PRODUCT_MAINTENANCE_VERSION%+%PRODUCT_BUILD_NUMBER%
 				IF !PRODUCT_CATEGORY! == jre (
 					SET REPRO_DIR=!REPRO_DIR!-!PRODUCT_CATEGORY!
 				)
 				IF NOT EXIST "!REPRO_DIR!" (
 					ECHO Third !REPRO_DIR! not exists
-					REM try folder for JDK-Latest defined in CreateSourceFolder.AdoptOpenJDK.ps1
-					SET REPRO_DIR=.\SourceDir\!PRODUCT_SKU!-Latest\!PACKAGE_TYPE!\!FOLDER_PLATFORM!\!ADOPT_NAME!
+					SET REPRO_DIR=.\SourceDir\!PRODUCT_SKU!!PRODUCT_MAJOR_VERSION!\!PACKAGE_TYPE!\!FOLDER_PLATFORM!\jdk-%PRODUCT_MAJOR_VERSION%.%PRODUCT_MINOR_VERSION%.%PRODUCT_MAINTENANCE_VERSION%.%PRODUCT_PATCH_VERSION%+%PRODUCT_BUILD_NUMBER%
+					IF !PRODUCT_CATEGORY! == jre (
+						SET REPRO_DIR=!REPRO_DIR!-!PRODUCT_CATEGORY!
+					)
 					IF NOT EXIST "!REPRO_DIR!" (
-						ECHO OpenJDK-Latest unnumbered !REPRO_DIR! does not exist
-						ECHO SOURCE Dir not found / failed
-						ECHO Listing directory :
-						dir /a:d /s /b /o:n SourceDir
-						GOTO FAILED
+						ECHO Fourth !REPRO_DIR! not exists
+						REM try folder for JDK-Latest defined in CreateSourceFolder.AdoptOpenJDK.ps1
+						SET REPRO_DIR=.\SourceDir\!PRODUCT_SKU!-Latest\!PACKAGE_TYPE!\!FOLDER_PLATFORM!\!ADOPT_NAME!
+						IF NOT EXIST "!REPRO_DIR!" (
+							ECHO OpenJDK-Latest unnumbered !REPRO_DIR! does not exist
+							ECHO SOURCE Dir not found / failed
+							ECHO Listing directory :
+							dir /a:d /s /b /o:n SourceDir
+							GOTO FAILED
+						)
```

Comment:
We probably have far more than 5 path to look for now.
Before there where 4 PATH + nightly :

jdk-%PRODUCT_MAJOR_VERSION%.%PRODUCT_MINOR_VERSION%.%PRODUCT_MAINTENANCE_VERSION%
jdk%PRODUCT_MAJOR_VERSION%u%PRODUCT_MAINTENANCE_VERSION%-b%PRODUCT_PATCH_VERSION%
jdk-%PRODUCT_MAJOR_VERSION%+%PRODUCT_PATCH_VERSION%
jdk-%PRODUCT_MAJOR_VERSION%.%PRODUCT_MINOR_VERSION%.%PRODUCT_MAINTENANCE_VERSION%+%PRODUCT_PATCH_VERSION%
!ADOPT_NAME!


Now 5 path :
jdk-%PRODUCT_MAJOR_VERSION%.%PRODUCT_MINOR_VERSION%.%PRODUCT_MAINTENANCE_VERSION%
jdk%PRODUCT_MAJOR_VERSION%u%PRODUCT_MAINTENANCE_VERSION%-b%PRODUCT_BUILD_NUMBER%
jdk-%PRODUCT_MAJOR_VERSION%+%PRODUCT_BUILD_NUMBER%
jdk-%PRODUCT_MAJOR_VERSION%.%PRODUCT_MINOR_VERSION%.%PRODUCT_MAINTENANCE_VERSION%+%PRODUCT_BUILD_NUMBER%
jdk-%PRODUCT_MAJOR_VERSION%.%PRODUCT_MINOR_VERSION%.%PRODUCT_MAINTENANCE_VERSION%.%PRODUCT_PATCH_VERSION%+%PRODUCT_BUILD_NUMBER%
and the nightly :
!ADOPT_NAME!

But now we must to look into 8 path ! + nitghly

![version](https://user-images.githubusercontent.com/45632394/98514704-3110e280-226a-11eb-9f4a-385aa6aa0e5f.png)

Reply:
Instead of trying to guess what the folder name inside the zip can be, it will be far more simple if the upstream job can pass the exact version folder name which is inside the zip ..

Reply:
And this 8 possible path is without taking into account that sometime there is a dash after "jdk" and sometime not
 and sometime there is a "u" before maintenance version number and sometime not and sometimes a "b" for build ..
Probably diff between 8 and 11+

Reply:
It's time to completly rewrite the logic before or just after this patch release